File: app/src/main/java/com/pedrocarrillo/redditclient/ui/singlePost/PostContractor.java
package com.pedrocarrillo.redditclient.ui.singlePost;

import com.pedrocarrillo.redditclient.ui.custom.BasePresenter;
import com.pedrocarrillo.redditclient.ui.custom.BaseView;

/**
 * @author pedrocarrillo.
 */

public interface PostContractor {

    interface View extends BaseView<PostContractor.Presenter> {

        void showInfo();

    }

    interface Presenter extends BasePresenter {

        void loadInfo(String permalink);

    }

}

File: app/src/main/java/com/pedrocarrillo/redditclient/ui/singlePost/PostContractor.kt
package com.pedrocarrillo.redditclient.ui.singlePost

import com.pedrocarrillo.redditclient.domain.Comment
import com.pedrocarrillo.redditclient.ui.custom.BasePresenter
import com.pedrocarrillo.redditclient.ui.custom.BaseView

/**
 * @author pedrocarrillo.
 */

interface PostContractor {

    interface View : BaseView<PostContractor.Presenter> {

        fun showInfo()
        fun showComment(comment : Comment)

    }

    interface Presenter : BasePresenter {

        fun loadInfo(permalink: String)

    }

}
